### PR TITLE
Add check for NftReadableKVState if spender is missing

### LINKED ISSUE #9792: Add check for NftReadableKVState if spender is missing
With the current implementation of NftReadableKVState if spender is not set for NFT it returns null. We should add a check if spender is not set

### PR DESCRIPTION
**Pull request by nickeynikolovv**
**Description**:
This PR adds a check in NftReadableKVState if spender is not set for NFT.
Also additional tests are added

**Related issue(s)**:

Fixes #9792 
**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

**Review by kselveliev — DISMISSED** [review]

File: hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/state/NftReadableKVStateTest.java [hunk 1]
```diff
@@ -169,6 +218,24 @@ private Nft setupNft(Optional<Long> timestamp) {
         return databaseNft;
     }
 
+    private Nft setupNftMissingSpender(Optional<Long> timestamp) {
+        Nft databaseNft = domainBuilder
+                .nft()
+                .customize(t -> t.tokenId(entity.getId())
+                        .serialNumber(NFT_ID.serialNumber()))
```
Comment (IvanKavaldzhiev) [inline hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/state/NftReadableKVStateTest.java:225]: Shouldn't we explicitly set the spender as null? Currently domainBuilder will generated a spender by default.

Comment (bilyana-gospodinova) [reply hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/state/NftReadableKVStateTest.java:225]: The `DomainBuilder` does not set a spender by default (it is null until set explicitly in the customizer). I just debugged the test to verify.

**Review by bilyana-gospodinova — APPROVED** [review]
LGTM

**Comment by steven-sheehy** [comment]
Might need a merge or rebase

**Review by steven-sheehy — APPROVED** [review]

**Pull request merged**